<a href="https://colab.research.google.com/github/DAVIDCRUZ0202/DS-Unit-1-Sprint-2-Statistics/blob/master/module1/LS_DS_121_Statistics_Probability_Assignment_David_Cruz.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img align="left" src="https://lever-client-logos.s3.amazonaws.com/864372b1-534c-480e-acd5-9711f850815c-1524247202159.png" width=200>
<br></br>
<br></br>

## *Data Science Unit 1 Sprint 2 Assignment 1*

# Apply the t-test to real data

Your assignment is to determine which issues have "statistically significant" differences between political parties in this [1980s congressional voting data](https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records). The data consists of 435 instances (one for each congressperson), a class (democrat or republican), and 16 binary attributes (yes or no for voting for or against certain issues). Be aware - there are missing values!

Your goals:

1. Load and clean the data (or determine the best method to drop observations when running tests)
2. Using hypothesis testing, find an issue that democrats support more than republicans with p < 0.01
3. Using hypothesis testing, find an issue that republicans support more than democrats with p < 0.01
4. Using hypothesis testing, find an issue where the difference between republicans and democrats has p > 0.1 (i.e. there may not be much of a difference)

Note that this data will involve *2 sample* t-tests, because you're comparing averages across two groups (republicans and democrats) rather than a single group against a null hypothesis.

In [1]:
### YOUR CODE STARTS HERE
import pandas as pd
import numpy as np
from scipy import stats

column_headers= ['party','handicapped-infants','water-project',
                          'budget','physician-fee-freeze', 'el-salvador-aid',
                          'religious-groups','anti-satellite-ban',
                          'aid-to-contras','mx-missile','immigration',
                          'synfuels', 'education', 'right-to-sue','crime','duty-free',
                          'south-africa']

df = pd.read_csv('https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data', names=column_headers)

print(df.shape)

df.head()

(435, 17)


party handicapped-infants water-project  ... crime duty-free south-africa
0  republican                   n             y  ...     y         n            y
1  republican                   n             y  ...     y         n            ?
2    democrat                   ?             y  ...     y         n            n
3    democrat                   n             y  ...     n         n            y
4    democrat                   y             y  ...     y         y            y

[5 rows x 17 columns]

In [2]:
df = df.replace({'y': 1, 'n': 0, '?': np.NaN})

df.head()

party  handicapped-infants  ...  duty-free  south-africa
0  republican                  0.0  ...        0.0           1.0
1  republican                  0.0  ...        0.0           NaN
2    democrat                  NaN  ...        0.0           0.0
3    democrat                  0.0  ...        0.0           1.0
4    democrat                  1.0  ...        1.0           1.0

[5 rows x 17 columns]

In [0]:
dem = df[df['party']=='democrat']

rep = df[df['party']=='republican']


In [0]:
# Using hypothesis testing, find an issue that Democrats support more than
# republicans with p < 0.01

1) Null Hypothesis: Republicans and Democrats support the budget bill about the same.

2) Alternative Hypothesis: Democrats support the budget bill at a higher rate than Republicans.

3) Confidence Level: 99%

In [29]:
stats.ttest_ind(rep['budget'], dem['budget'], nan_policy='omit')

Ttest_indResult(statistic=-23.21277691701378, pvalue=2.0703402795404463e-77)

Conclusion: With a t statistic of -23.2127, and a P-Value of 2.0703 to the -77th power, i reject the null hypothesis, and suggest the alternative hypothesis that democrats support the budget bill at a higher rate than republicans.

In [10]:
rep['budget'].value_counts()

0.0    142
1.0     22
Name: budget, dtype: int64

In [11]:
dem['budget'].value_counts()

1.0    231
0.0     29
Name: budget, dtype: int64

In [0]:
### Using hypothesis testing, find an issue that republicans support more than 
### democrats with p < 0.01

1) Null Hypothesis: Republicans support the religious groups bill at the same rate as Democrats.

2) Alternative Hypothesis: Republicans support the religious groups bill at some other rate higher than Democrats.

3) Confidence Level: 99%

In [15]:
stats.ttest_ind(rep['religious-groups'], dem['religious-groups'], nan_policy='omit')

Ttest_indResult(statistic=9.737575825219457, pvalue=2.3936722520597287e-20)

Conclusion: With a t statistic of -23.2127, and a P-Value of 2.0703 to the -77th power, i reject the null hypothesis, and suggest the alternative hypothesis that republicans and democrats support the budget bill at some other rate. (Dems support more because of the negative t-statistic)



In [26]:
rep['religious-groups'].value_counts()

1.0    149
0.0     17
Name: religious-groups, dtype: int64

In [25]:
dem['religious-groups'].value_counts()

0.0    135
1.0    123
Name: religious-groups, dtype: int64

**** Part 4****
Using hypothesis testing, find an issue where the difference between republicans and democrats has p > 0.1 (i.e. there may not be much of a difference)

1) Null Hypothesis: Republicans support the education bill at a level similar to Democrats.

2) Alternative Hypothesis: Republicans and Democrats support the education bill at some other rate.

3) Confidence Level: 90%

In [5]:
stats.ttest_ind(dem['water-project'], rep['water-project'], nan_policy='omit')

Ttest_indResult(statistic=-0.08896538137868286, pvalue=0.9291556823993485)

Conclusion: with a t statistic of -.08 and a p-value of .9291, I fail to reject the null hypothesis which suggests that republicans support the education bill at a level similar to democrats.

## Stretch Goals:

1. Refactor your code into functions so it's easy to rerun with arbitrary variables
2. Work on Performing a T-test without using Scipy in order to get "under the hood" and learn more thoroughly about this topic.
### Start with a 1-sample t-test
 - Establish the conditions for your test 
 - [Calculate the T Statistic](https://blog.minitab.com/hs-fs/hubfs/Imported_Blog_Media/701f9c0efa98a38fb397f3c3ec459b66.png?width=247&height=172&name=701f9c0efa98a38fb397f3c3ec459b66.png) (You'll need to omit NaN values from your sample).
 - Translate that t-statistic into a P-value. You can use a [table](https://www.google.com/search?q=t+statistic+table) or the [University of Iowa Applet](https://homepage.divms.uiowa.edu/~mbognar/applets/t.html)

 ### Then try a 2-sample t-test
 - Establish the conditions for your test 
 - [Calculate the T Statistic](https://lh3.googleusercontent.com/proxy/rJJ5ZOL9ZDvKOOeBihXoZDgfk7uv1YsRzSQ1Tc10RX-r2HrRpRLVqlE9CWX23csYQXcTniFwlBg3H-qR8MKJPBGnjwndqlhDX3JxoDE5Yg) (You'll need to omit NaN values from your sample).
 - Translate that t-statistic into a P-value. You can use a [table](https://www.google.com/search?q=t+statistic+table) or the [University of Iowa Applet](https://homepage.divms.uiowa.edu/~mbognar/applets/t.html)

 ### Then check your Answers using Scipy!